### ISSUE TITLE
🐞 Generate sources fails when dagger module is in .dagger directory

### ISSUE THREAD

**Issue by nicholasjackson** [dagger/dagger#7078]
### What is the issue?

With dagger v0.11.0, when running `dagger develop` and the module is in a folder called `.dagger` the package `main/internal/telemetry` is not added to the `dagger.gen.go` fille. The imports are as follows.

```
import (
	"context"
	"encoding/json"
	"fmt"
	"log/slog"
	"os"

	"main/internal/dagger"

	"go.opentelemetry.io/otel"
	"go.opentelemetry.io/otel/attribute"
	"go.opentelemetry.io/otel/sdk/resource"
	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
	"go.opentelemetry.io/otel/trace"
)
```

### Dagger version

dagger v0.11.0

### Steps to reproduce

The following repository contains a dagger module at `.dagger` running `dagger develop` on this folder will incorrectly generate the sources.

https://github.com/jumppad-labs/jumppad/tree/52c3ff885cf616650ef997de36768663e8a4048f

**Comment by jedevc** [comment]
Aha, this looks like the same issue I discovered while working on https://github.com/dagger/dagger/pull/7194 - the fix is part of that PR in [`6fdd040` (#7194)](https://github.com/dagger/dagger/pull/7194/commits/6fdd040c71288ccb3385829b29cd25ac91f8f79d).

**Comment by jedevc** [comment]
Not 100% sure why this only seems to appear in a small sub-section of cases, but it definitely needs fixing :scream:
